### ISSUE TITLE
[prefer-readonly-signals] New rule for checking that signals are defined as readonly

### ISSUE THREAD

**Issue by m-thalmann** [angular-eslint/angular-eslint#1719]
**Description and reproduction of the issue**

The new Angular signals should only be updated using the `.set()` or `.update()` function, never overwritten.
For this reason I propose a new rule which checks that all signals are marked as readonly.

**Incorrect examples**

```ts
myCounter = signal(0);
anotherCounter = input(0);
doubleCounter = computed(() => this.myCounter() * 2);
```

**Correct examples**

```ts
readonly myCounter = signal(0);
// ...
```

**Comment by reduckted** [comment]
There's a few other signal-creating functions that should be included by this rule:

* [`contentChild`](https://angular.dev/api/core/contentChild)
* [`contentChildren`](https://angular.dev/api/core/contentChildren)
* [`input`](https://angular.dev/api/core/input)
* [`model`](https://angular.dev/api/core/model)
* [`output`](https://angular.dev/api/core/output)
* [`viewChild`](https://angular.dev/api/core/viewChild)
* [`viewChildren`](https://angular.dev/api/core/viewChildren)

**Comment by reduckted** [comment]
> * output

🤦‍♂️ `output()` does not create a signal, so it's not relevant to this new rule. I've logged #1871 to add it to the existing `prefer-output-readonly` rule.

**Comment by P4** [comment]
@reduckted do you think [`toSignal`](https://angular.dev/api/core/rxjs-interop/toSignal) from RxJS interop package should be included too?

**Comment by reduckted** [comment]
Indeed! Your comment in #1876 reminded me about it! 😆

**Comment by JamesHenry** [comment]
This was included in the new `prefer-signals` rule in v19:

https://github.com/angular-eslint/angular-eslint/blob/af5386760858602db8a70f70ee06fc5991850c6f/packages/eslint-plugin/docs/rules/prefer-signals.md
